packages/cdktf-cli/lib/cdktf-config.ts

```diff
@@ -0,0 +1,89 @@
+import * as fs from "fs";
+import { Language } from "@cdktf/provider-generator";
+import { findFileAboveCwd } from "../bin/cmds/helper/utilities";
+import { Errors } from "./errors";
+import path from "path";
+import { logger } from "./logging";
+import { CONFIG_DEFAULTS } from "@cdktf/provider-generator/lib/config";
+
+export type ProviderDependencySpec = {
+  name: string;
+  source: string;
+  version?: string;
+  namespace?: string;
+};
+
+// TODO: move this to some common package from where it is used by @cdktf/provider-generator, cdktf, and cdktf-cli
```

Comment:
Nit: sounds good, let's move it and refactor all the other places at the same time, because I wrote this quite a few times already :D

Reply:
Yeah sounds like a good idea

Reply:
added https://github.com/hashicorp/terraform-cdk/issues/1814 and linked it in the comment